smithy-typescript-codegen/build.gradle.kts

```diff
@@ -18,6 +18,7 @@ extra["displayName"] = "Smithy :: Typescript :: Codegen"
 extra["moduleName"] = "software.amazon.smithy.typescript.codegen"
 
 dependencies {
+    api("software.amazon.smithy:smithy-aws-traits:[1.7.2, 2.0[")
```

Comment:
I think we're looking to avoid taking this dependency in this package. It'll require changing the resolution of the `serviceId` for creating a `TraceMetadata` in the `CodegenVisitor`, but seems like the right call.

Reply:
Yes, that makes sense. Updated to remove the dependency and this will now just use the service name.